File: oa_app/lib/features/auth/models/login_response.dart
class LoginResponse {
  final String accessToken;
  final int userId;
  final String username;
  final String realName;
  final String? avatarUrl;
  final List<String> roles;
  final List<String> permissions;

  const LoginResponse({
    required this.accessToken,
    required this.userId,
    required this.username,
    required this.realName,
    this.avatarUrl,
    required this.roles,
    required this.permissions,
  });

  factory LoginResponse.fromJson(Map<String, dynamic> json) {
    return LoginResponse(
      accessToken: json['accessToken'] as String,
      userId: json['userId'] as int,
      username: json['username'] as String,
      realName: json['realName'] as String,
      avatarUrl: json['avatarUrl'] as String?,
      roles: List<String>.from(json['roles'] as List? ?? []),
      permissions: List<String>.from(json['permissions'] as List? ?? []),
    );
  }
}

File: OA-System-AI/user-service/src/main/java/com/oa/user/dto/LoginResponse.java
package com.oa.user.dto;

import com.oa.user.security.LoginUserDetails;

import java.util.List;

public class LoginResponse {

    private String accessToken;
    private String tokenType;
    private long expiresIn;
    private Long userId;
    private Long deptId;
    private String username;
    private List<String> roles;
    private List<String> permissions;

    public static LoginResponse from(LoginUserDetails userDetails, String token, long expiresIn) {
        LoginResponse response = new LoginResponse();
        response.setAccessToken(token);
        response.setTokenType("Bearer");
        response.setExpiresIn(expiresIn);
        response.setUserId(userDetails.getUserId());
        response.setDeptId(userDetails.getDeptId());
        response.setUsername(userDetails.getUsername());
        response.setRoles(userDetails.getRoles());
        response.setPermissions(userDetails.getPermissions());
        return response;
    }

    public String getAccessToken() {
        return accessToken;
    }

    public void setAccessToken(String accessToken) {
        this.accessToken = accessToken;
    }

    public String getTokenType() {
        return tokenType;
    }

    public void setTokenType(String tokenType) {
        this.tokenType = tokenType;
    }

    public long getExpiresIn() {
        return expiresIn;
    }

    public void setExpiresIn(long expiresIn) {
        this.expiresIn = expiresIn;
    }

    public Long getUserId() {
        return userId;
    }

    public void setUserId(Long userId) {
        this.userId = userId;
    }

    public Long getDeptId() {
        return deptId;
    }

    public void setDeptId(Long deptId) {
        this.deptId = deptId;
    }

    public String getUsername() {
        return username;
    }

    public void setUsername(String username) {
        this.username = username;
    }

    public List<String> getRoles() {
        return roles;
    }

    public void setRoles(List<String> roles) {
        this.roles = roles;
    }

    public List<String> getPermissions() {
        return permissions;
    }

    public void setPermissions(List<String> permissions) {
        this.permissions = permissions;
    }
}
